DANS-KNAW/dd-pre-publish-workflow
# DRAFT Version of Dataverse Workflow POC

Fixes DD-39 - POC handle licence field (WIP)

* See: https://drivenbydata.atlassian.net/browse/DD-39

#### Where should the reviewer start?

Note:   
I was not able to fix the issue that first an Ok() has to be sent before the resume request is done. 
Therefore the dataset is always locked when publishing. 

#### How should this be manually tested?

1. Check out this PR
2. Upload Workflow  
`curl http://localhost:8080/api/admin/workflows -X POST --data-binary @{absolute path}/dd-easy-worflows-poc/src/main/scala/nl/knaw/dans/dd/wf/workflows/license-workflow.json -H "Content-type: application/json"`
3. Set as default Pre-Publish Workflow:  
`curl http://localhost:8080/api/admin/workflows/default/PrePublishDataset -X PUT -d 1`
4. Enable the license metadatablock in Dataverse
5. Create a new dataset with a licence and an access level
6. Publish
7. Check terms tab.
